src/dds/build/plan/library.cpp

```diff
@@ -56,12 +59,27 @@ library_plan library_plan::create(const library_root&             lib,
                 template_sources.push_back(sfile);
             } else {
                 assert(sfile.kind == source_kind::header);
+                header_sources.push_back(sfile);
             }
         }
     }
 
+    auto include_dir = lib.include_source_root();
+    if (include_dir.exists()) {
+        auto all_sources = include_dir.collect_sources();
+        for (const auto& sfile : all_sources) {
+            assert(sfile.kind == source_kind::header || sfile.kind == source_kind::header_template);
```

Comment:
I think it would actually make sense to not do an assertion at all here, because AFAICT, this is the only location that DDS checks the file kind of files under `include/`. I made this change, instead having a `dds_log(warn, ...)` if the header is not a header or header template.

Reply:
It's actually checked: sdist/library/root.cpp's `collect_pf_sources` has 

```c++
                dds_log(warn,
                        "Source file in `include` will not be compiled: {}",
                        info.path.string());
```